src/main/java/appeng/fluids/parts/PartFluidInterface.java

```diff
@@ -0,0 +1,189 @@
+import appeng.fluids.helper.IFluidInterfaceHost;
+import appeng.helpers.Reflected;
+import appeng.items.parts.PartModels;
+import appeng.parts.PartBasicState;
+import appeng.parts.PartModel;
+import appeng.util.Platform;
+
+
+public class PartFluidInterface extends PartBasicState implements IGridTickable, IStorageMonitorable, IFluidInterfaceHost
+{
+	public static final ResourceLocation MODEL_BASE = new ResourceLocation( AppEng.MOD_ID, "part/interface_base" );
+
+	@PartModels
+	public static final PartModel MODELS_OFF = new PartModel( MODEL_BASE, new ResourceLocation( AppEng.MOD_ID, "part/interface_off" ) );
+
+	@PartModels
+	public static final PartModel MODELS_ON = new PartModel( MODEL_BASE, new ResourceLocation( AppEng.MOD_ID, "part/interface_on" ) );
+
+	@PartModels
+	public static final PartModel MODELS_HAS_CHANNEL = new PartModel( MODEL_BASE, new ResourceLocation( AppEng.MOD_ID, "part/interface_has_channel" ) );
+
+	private final DualityFluidInterface duality = new DualityFluidInterface( this.getProxy(), this );
+
+	@Reflected
+	public PartFluidInterface( final ItemStack is )
+	{
+		super( is );
+	}
+
+	@Override
+	public DualityFluidInterface getDualityFluidInterface()
+	{
+		return duality;
+	}
+
+	@MENetworkEventSubscribe
+	public void stateChange( final MENetworkChannelsChanged c )
+	{
+		this.duality.notifyNeighbors();
+	}
+
+	@MENetworkEventSubscribe
+	public void stateChange( final MENetworkPowerStatusChange c )
+	{
+		this.duality.notifyNeighbors();
+	}
+
+	@Override
+	public void getBoxes( final IPartCollisionHelper bch )
+	{
+		bch.addBox( 2, 2, 14, 14, 14, 16 );
+		bch.addBox( 5, 5, 12, 11, 11, 14 );
+	}
+
+	@Override
+	public void gridChanged()
+	{
+		this.duality.gridChanged();
+	}
+
+	@Override
+	public void readFromNBT( final NBTTagCompound data )
+	{
+		super.readFromNBT( data );
+		this.duality.readFromNBT( data );
+	}
+
+	@Override
+	public void writeToNBT( final NBTTagCompound data )
+	{
+		super.writeToNBT( data );
+		this.duality.writeToNBT( data );
+	}
+
+	@Override
+	public float getCableConnectionLength( AECableType cable )
+	{
+		return 4;
+	}
+
+	@Override
+	public boolean onPartActivate( final EntityPlayer p, final EnumHand hand, final Vec3d pos )
+	{
+		if( p.isSneaking() )
+		{
+			return false;
+		}
+
+		if( Platform.isServer() )
+		{
+			Platform.openGUI( p, this.getTileEntity(), this.getSide(), GuiBridge.GUI_FLUID_INTERFACE );
+		}
+
+		return true;
+	}
+
+	@Override
+	public <T extends IAEStack<T>> IMEMonitor<T> getInventory( IStorageChannel<T> channel )
+	{
+		return this.duality.getInventory( channel );
+	}
+
+	@Override
+	public TickingRequest getTickingRequest( final IGridNode node )
+	{
+		return this.duality.getTickingRequest( node );
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( final IGridNode node, final int ticksSinceLastCall )
+	{
+		return this.duality.tickingRequest( node, ticksSinceLastCall );
+	}
+
+	@Override
+	public EnumSet<EnumFacing> getTargets()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "Set" rather than the implementation "EnumSet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1319)